### PR TITLE
Supportpluginregister

### PR DESCRIPTION
**Pull request by orthagonal**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -46,12 +46,26 @@ class Logger {
...
+          plugin = require(renderer);
+        } else {
+          plugin = renderer;
+        }
+        // each renderer can either be a callable function or an
+        // object having a 'register' and 'render' function
+        if (plugin.register) {
+          plugin.register(this.config.renderOptions[renderName], (err) => {
+            if (err) {
+              return console.log(err);
```
Comment (jgallen23) [inline index.js]: this needs to throw

**Comment by jgallen23** [comment]
I'll be honest.  I have no idea what this does

**Comment by orthagonal** [comment]
Some logr plugins will want to initialize some things up-front before they can handle log calls.  Logr-slack uses 'register' to export its internal methods. hapi-slack in turn is able to decorate a hapi server with those methods.

File: index.js [hunk 2]
```diff
@@ -34,57 +33,104 @@ let defaults = {
...
       json: require('./lib/json'),
       cli: require('./lib/cli'),
     };
-    if (this.config.plugins) {
-      _.each(options.plugins, (renderFunction, renderName) => {
-        if (typeof renderFunction === 'string') {
-          this.renderers[renderName] = require(renderFunction);
-          return;
+    if (this.config.additionalRenderers) {
+      _.each(options.additionalRenderers, (renderer, renderName) => {
```
Comment (jgallen23) [inline index.js]: can we move this to a new method to clean this up a bit?

File: index.js [hunk 3]
```diff
@@ -34,57 +33,104 @@ let defaults = {
...
-  filterMatch(filter, tags) {
-    if (filter === null) {
-      return true;
-    }
-    return filter.some((filterTag) => {
-      return tags.indexOf(filterTag) !== -1;
+    this.plugins = {};
+    const pluginList = this.config.plugins ? this.config.plugins : [];
+    // load the default plugin:
+    pluginList.push('./lib/defaultPlugin');
```
Comment (jgallen23) [inline index.js]: same as above, we should move to an `initPlugins` method

**Pull request closed without merging**